core/src/test/java/org/apache/calcite/test/enumerable/EnumerableRepeatUnionTest.java

```diff
@@ -45,6 +45,17 @@
  */
 class EnumerableRepeatUnionTest {
 
+  @Test void testGenerateNumbersUsingSql() {
```

Comment:
The new tests added here are basically testing recursive queries end-to-end. For end-to-end tests it is better to use the .iq files so consider moving the tests in `recursive_queries.iq` and remove them from here unless there is a specific reason to have them here.